Frontend/ui-driver/src/Screens/RegistrationScreen/View.purs

```diff
@@ -361,6 +375,45 @@ messageView push state =
           ]
       ]
 
+enterReferralCode :: forall w . (Action -> Effect Unit) -> ST.RegistrationScreenState -> PrestoDOM (Effect Unit) w
+enterReferralCode push state =
+  let allStepsCompleted = all (_ == COMPLETED) [ state.data.vehicleDetailsStatus, state.data.drivingLicenseStatus, state.data.permissionsStatus ]
+    in linearLayout
+            [ width MATCH_PARENT
+            , height WRAP_CONTENT
+            , orientation HORIZONTAL
+            , stroke $ "1," <> Color.grey900
+            , cornerRadius 4.0
+            , padding $ Padding 10 10 10 10
+            ][  textView $
+                [ width MATCH_PARENT
+                , height WRAP_CONTENT
+                , color if allStepsCompleted then Color.black900 else Color.greyTextColor
+                , text if state.props.referralCodeSubmitted then (getString REFERRAL_APPLIED) else (getString HAVE_A_REFERRAL_CODE)
```

Comment:
`, text $ getString if state.props.referralCodeSubmitted then REFERRAL_APPLIED else HAVE_A_REFERRAL_CODE`

This might look cleaner